scipy/stats/_binomtest.py

```diff
@@ -313,16 +312,59 @@ def binomtest(k, n, p=0.5, alternative='two-sided'):
             # special case as shortcut, would also be handled by `else` below
             pval = 1.
         elif k < p * n:
-            i = np.arange(np.ceil(p * n), n+1)
-            y = np.sum(binom.pmf(i, n, p) <= d*rerr, axis=0)
+            y = n-_binary_search_for_binom_tst(lambda x1: -binom.pmf(x1, n, p),
+                                               -d*rerr, np.ceil(p * n), n)
             pval = binom.cdf(k, n, p) + binom.sf(n - y, n, p)
         else:
-            i = np.arange(np.floor(p*n) + 1)
-            y = np.sum(binom.pmf(i, n, p) <= d*rerr, axis=0)
+            y = _binary_search_for_binom_tst(lambda x1: binom.pmf(x1, n, p),
+                                             d*rerr, 0, np.floor(p * n) + 1,
+                                             True)+1
```

Comment:
```suggestion
                                             d*rerr, 0, np.floor(p * n) + 1)+1
```
IIUC the ascending order argument is no longer needed

Reply:
Yes, good catch. Removed.